optuna/exceptions.py

```diff
@@ -13,14 +13,40 @@ class TrialPruned(OptunaError):
 
     Example:
 
-        .. code::
-
-            >>> def objective(trial):
-            >>>     ...
-            >>>     for step in range(n_train_iter):
-            >>>         ...
-            >>>         if trial.should_prune():
-            >>>             raise TrailPruned()
+        .. testsetup::
+
+            import numpy as np
+            from sklearn.model_selection import train_test_split
+
+            np.random.seed(seed=0)
+            X = np.random.randn(200).reshape(-1, 1)
+            y = np.where(X[:, 0] < 0.5, 0, 1)
+            X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+            classes = np.unique(y)
+
+        .. testcode::
+
+            import optuna
+            from sklearn.linear_model import SGDClassifier
+
+            def objective(trial):
+                alpha = trial.suggest_uniform('alpha', 0.0, 1.0)
+                clf = SGDClassifier(alpha=alpha)
+                n_train_iter = 100
+
+                for step in range(n_train_iter):
+                    clf.partial_fit(X_train, y_train, classes=classes)
+
+                    intermediate_value = 1.0 - clf.score(X_test, y_test)
```

Comment:
I think we can maximize accuracy by using `direction` argument of `optuna.create_study`.
```suggestion
                    intermediate_value = clf.score(X_test, y_test)
```